twake/backend/node/src/core/platform/services/push/connectors/fcm/service.ts

```diff
@@ -0,0 +1,47 @@
+import { option } from "yargs";
+import { logger } from "../../../../framework";
+import { PushConfiguration, PushMessageNotification, PushMessageOptions } from "../../types";
+import { PushConnector } from "../connector";
+
+export default class FcmPushConnector implements PushConnector {
+  name = "FcmPushConnector";
+
+  constructor(readonly configuration: PushConfiguration["fcm"]) {}
+
+  async push(
+    devices: string[],
+    notification: PushMessageNotification,
+    options: PushMessageOptions,
+  ): Promise<void> {
+    const firebaseEndpoint = this.configuration.endpoint;
+    const firebaseApiKey = this.configuration.key;
+
+    for (const deviceIdentifier of devices) {
```

Comment:
I do not know how FCM works, but here you are looping over devices and sending as many HTTP requests as there are devices. I see that the `pushMessages` contains an array of `registration_ids` where you put only one itel when you can probably put more than one.

Reply:
Oups, it's an error my bad, thanks